docs/DeveloperGuide.md

```diff
@@ -243,54 +244,250 @@ a. Include those libraries in the repo (this bloats the repo size)<br>
 b. Require developers to download those libraries manually (this creates extra work for developers)<br>
```

Comment:
It may be more practical to give a few examples of interesting test cases and explain the thinking behind how you arrived at those test cases, rather than try to give a comprehensive list of all test cases.